File: Examples/TAKUUIDSample/TAKViewController.m
#import "TAKViewController.h"
#import <TAKUUID/TAKUUID.h>

@implementation TAKViewController

- (IBAction)remove {
  BOOL result = [[TAKUUIDStorage sharedInstance] remove];
  NSLog(@"remove => %d", result);
  NSLog(@"lastErrorStatus = %d", (int)[TAKUUIDStorage sharedInstance].lastErrorStatus);
}

- (IBAction)findOrCreate {
  NSString *UUIDString = [[TAKUUIDStorage sharedInstance] findOrCreate];
  NSLog(@"UUID = %@", UUIDString);
  NSLog(@"lastErrorStatus = %d", (int)[TAKUUIDStorage sharedInstance].lastErrorStatus);
}

- (IBAction)renew {
  NSString *UUIDString = [[TAKUUIDStorage sharedInstance] renew];
  NSLog(@"UUID = %@", UUIDString);
  NSLog(@"lastErrorStatus = %d", (int)[TAKUUIDStorage sharedInstance].lastErrorStatus);
}

- (IBAction)updateAttrAccessible {
  NSString *UUIDString = [[TAKUUIDStorage sharedInstance] findOrCreate];
  NSLog(@"UUID = %@", UUIDString);
  NSLog(@"lastErrorStatus = %d", (int)[TAKUUIDStorage sharedInstance].lastErrorStatus);
  
  [[TAKUUIDStorage sharedInstance] updateAttrAccessible];
  
  UUIDString = [[TAKUUIDStorage sharedInstance] findOrCreate];
  NSLog(@"UUID = %@", UUIDString);
}

@end

File: Examples/TAKUUIDSample/TAKViewController.swift
//
//  TAKViewController.swift
//
//  Created by Takahiro Oishi
//  Copyright (c) 2016年 Takahiro Ooishi. All rights reserved.
//

import UIKit

final class TAKViewController: UIViewController {
  override func viewDidLoad() {
    guard let accessGroup = Bundle.main.object(forInfoDictionaryKey: "KeychainAccessGroup") as? String else { return }
    TAKUUIDStorage.sharedInstance().accessGroup = accessGroup
    TAKUUIDStorage.sharedInstance().migrate()
  }
  
  @IBAction func remove() {
    let result = TAKUUIDStorage.sharedInstance().remove()
    print("remove = \(result)")
    print("lastErrorStatus = \(TAKUUIDStorage.sharedInstance().lastErrorStatus)")
  }
  
  @IBAction func findOrCreate() {
    guard let uuid = TAKUUIDStorage.sharedInstance().findOrCreate() else { return }
    print("uuid = \(uuid)")
    print("lastErrorStatus = \(TAKUUIDStorage.sharedInstance().lastErrorStatus)")
  }
  
  @IBAction func renew() {
    guard let uuid = TAKUUIDStorage.sharedInstance().renew() else { return }
    print("uuid = \(uuid)")
    print("lastErrorStatus = \(TAKUUIDStorage.sharedInstance().lastErrorStatus)")
  }
}
